content/en/docs/languages/go/instrumentation.md

```diff
@@ -690,12 +711,37 @@ func init() {
 
 ### Adding attributes
 
+Attributes are keys and values that are applied as metadata to your metrics and
+are useful for aggregating, filtering, and grouping metrics. Attributes can be
+added at instrument creation, or at any other time during the lifecycle of an
+instrument.
+
+Generally speaking, it is advisable to add attributes during instrumentation
+if you can. There is memory overhead when adding them after creation, and in
+some scenarios this can degrade performance.
+
 You can add Attributes by using the
 [`WithAttributeSet`](https://pkg.go.dev/go.opentelemetry.io/otel/metric#WithAttributeSet)
 or
 [`WithAttributes`](https://pkg.go.dev/go.opentelemetry.io/otel/metric#WithAttributes)
 options.
 
+For example, here's how you could add descriptive attributes during creation:
+
+```go
+counter, err := meter.Int64UpDownCounter(
+	"app.my-counter",
+	metric.WithDescription("My fancy counter."),
+	metric.WithAttributes(
+		attribute.String("my-important-key", "my-important-value"),
```

Comment:
mm.. not sure if this is a good practice.. if the key and value are known ahead, are they better fitted as instrumentation scope attributes, set on the Meter itself, so they won't take up aggregation space, and are more efficiently transported over otlp too?

(Please ignore if this comment does not makes sense. I have limited Go experience)

Reply:
Yeah, I was thinking maybe there's three scenarios here:

- Common attributes at the meter level (perhaps resource level?)
- Attributes you want per instrument
- Attributes you need to add when you record a measurement

The first category is the weirdest one to me. But also there's I think a different mindset eto attributes for metrics that I'm not considering here.

Reply:
The first category is already well defined - https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#get-a-meter
_[since 1.13.0] attributes: Specifies the instrumentation scope attributes to associate with emitted telemetry._

The 2nd one - instrument level attributes - they don't exist in the spec from what I can tell.

Reply:
> The 2nd one - instrument level attributes - they don't exist in the spec from what I can tell.

Correct. Have you checked if such code compiles?

Reply:
Reverted that portion